Cesium3DTilesSelection/include/Cesium3DTilesSelection/WebMapTileServiceRasterOverlay.h

```diff
@@ -0,0 +1,161 @@
+#pragma once
+
+#include "Library.h"
+#include "RasterOverlay.h"
+
+#include <CesiumAsync/IAssetRequest.h>
+#include <CesiumGeometry/QuadtreeTilingScheme.h>
+#include <CesiumGeospatial/Ellipsoid.h>
+#include <CesiumGeospatial/GlobeRectangle.h>
+#include <CesiumGeospatial/Projection.h>
+
+#include <functional>
+#include <memory>
+
+namespace Cesium3DTilesSelection {
+
+class CreditSystem;
+
+/**
+ * @brief Options for tile map service accesses.
+ */
+struct WebMapTileServiceRasterOverlayOptions {
+
+  /**
+   * @brief The MIME type for images to retrieve from the server.
+   */
+  std::optional<std::string> format;
+
+  /**
+   * @brief The subdomains to use for the <code>{s}</code> placeholder in the URL template.
+   *
+   * If this parameter is a single string, each character in the string is a
+   * subdomain. If it is an array, each element in the array is a subdomain.
+   */
+  std::optional<std::variant<std::string, std::vector<std::string>>> subdomains;
+
+  /**
+   * @brief A credit for the data source, which is displayed on the canvas.
+   */
+  std::optional<std::string> credit;
+
+  /**
+   * @brief The layer name for WMTS requests.
+   */
+  std::optional<std::string> layer;
+
+  /**
+   * @brief The style name for WMTS requests.
+   */
+  std::optional<std::string> style;
+
+  /**
+   * @brief The identifier of the TileMatrixSet to use for WMTS requests.
+   */
+  std::optional<std::string> tileMatrixSetID;
+
+  /**
+   * @brief A list of identifiers in the TileMatrix to use for WMTS requests,
+   * one per TileMatrix level.
+   */
+  std::optional<std::vector<std::string>> tileMatrixLabels;
```

Comment:
For parameters that are required, like this one, it's probably clearer not to use `std::optional`. That won't save you from needing to check for an empty string (assuming an empty string is an invalid tileMatrixLabel?) but it does make the usage a little more obvious to users.

Reply:
do you mean tileMatrixSetID?